config/include.go

```diff
@@ -659,6 +662,26 @@ func deepMergeDependencyBlocks(targetDependencies []Dependency, sourceDependenci
 	return combinedDeps, nil
 }
 
+// Merge list argument preserving empty lists.
+//
+// If child provides an explicit empty list, we store a non-nil empty slice
+// to represent "clear the list", instead of a nil slice.
+func mergeListArgPreserveEmpty(childListArg *[]string, parentListArg **[]string) {
+	if childListArg == nil {
+		return
+	}
+
+	if len(*childListArg) == 0 {
+		empty := make([]string, 0)
+		*parentListArg = &empty
+
+		return
+	}
+
+	copied := append([]string(nil), *childListArg...)
+	*parentListArg = &copied
+}
```

Comment:
Either this function is wrong, or the name + docstring is. This doesn't merge the two lists under any circumstances; it either:
- replaces the parent list with a new empty list if the child list is empty (couldn't we just use the existing one instead of allocating a new one?)
- replaces the parent list with the non-empty child list
- leaves the parent list alone if the child list is nil

none of these are a "merge"